### PR TITLE
Make the desktop Rewards page responsive

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/16619

### Implementation Notes

- Usage of `chrome.braveRewards` has been removed from the Rewards page. The message handler will now handle displaying of grant captchas in the Rewards panel on desktop.
- Unused message handlers "claimPromotion" and "attestPromotion" have been removed.
- Other minor message handler cleanups.
- Removed unused assets.
- Removed unused actions.
- Converted component code that directly called "chrome.send" into actions.
- Moved ads-specific styles into `adsBox.style.ts`
- Introduced hooks for accessing actions and Redux state.
- Introduced `LayoutManager` for setting the current layout based on the window width.
- Moved root-level files (e.g. `utils.tsx`) into associated folders.
- Newer, "mobile-style" version of the pending contributions modal replaces the old desktop modal.
- `settingsPage.tsx` has been rewritten using hooks.
- State caching and default state has been moved into reducers.

### UI differences in narrow-mode

- Promos are not displayed.
- Manage wallet button is not displayed.
- Ads history button is not displayed.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Rewards page should adapt to the browser's window size, with a break into "narrow" mode at 880px.

The following UI differences in "narrow" mode are expected:

- Sidebar promos are not displayed.
- Manage wallet button is not displayed.
- Ads history button is not displayed.

Due to refactoring, the following features on the Rewards page should be regression tested:

- Claiming a grant (UGP is sufficient)
- Viewing (and removing) pending contributions.
- Connecting a Gemini external wallet.
- Connecting an external wallet, where an error modal is expected.

### CODE REVIEW

File: components/brave_rewards/resources/page/brave_rewards_page.tsx [hunk 1]
```diff
@@ -200,136 +184,131 @@ function reconcileComplete (properties: {type: number, result: number}) {
...
 }
 
 function unblindedTokensReady () {
-  getActions().getBalance()
+  actions.getBalance()
 }
 
-function monthlyReport (properties: { result: number, month: number, year: number, report: Rewards.MonthlyReport}) {
-  getActions().onMonthlyReport(properties)
+function monthlyReport (properties: any) {
```
Comment (emerick) [inline components/brave_rewards/resources/page/brave_rewards_page.tsx:212]: Should we define a type for this or it's not worth it?

File: components/brave_rewards/resources/page/components/pageWallet.tsx [hunk 2]
```diff
@@ -35,8 +41,18 @@ interface State {
...
 
+function constructBackupString (backupKey: string) {
+  return 'Brave Wallet Recovery Key\n' +
+    `Date created: ${new Date(Date.now()).toLocaleDateString()}\n\n` +
+    `Recovery Key: ${backupKey}\n\n` +
+    'Note: This key is not stored on Brave servers. This key is your only ' +
+    'method of recovering your Brave wallet. Save this key in a safe place, ' +
+    'separate from your Brave browser. Make sure you keep this key private, ' +
+    'or else your wallet will be compromised.'
+}
```
Comment (emerick) [inline components/brave_rewards/resources/page/components/pageWallet.tsx:52]: Is it intentional that we're not localizing this?

File: components/brave_rewards/resources/page/components/pageWallet.tsx [hunk 3]
```diff
@@ -107,16 +124,16 @@ class PageWallet extends React.Component<Props, State> {
     if (document.location) {
       const win = window.open(document.location.href)
       if (win) {
-        win.document.body.innerText = utils.constructBackupString(backupKey) // this should be text, not HTML
+        win.document.body.innerText = constructBackupString(backupKey) // this should be text, not HTML
```
Comment (emerick) [inline components/brave_rewards/resources/page/components/pageWallet.tsx:127]: Should `utils.constructBackupString` go away or is it still used?

File: browser/ui/webui/brave_rewards_page_ui.cc [hunk 4]
```diff
@@ -808,46 +778,40 @@ void RewardsDOMHandler::ClaimPromotion(const base::Value::List& args) {
...
-    const std::string& promotion_id,
-    const ledger::type::Result result,
-    ledger::type::PromotionPtr promotion) {
   if (!IsJavascriptAllowed()) {
     return;
   }
 
   base::Value::Dict promotion_dict;
-  promotion_dict.Set("promotionId", promotion_id);
+  promotion_dict.Set("promotionId", promotion->id);
```
[the remark this answers is not in the document]
Comment (zenparsing) [reply browser/ui/webui/brave_rewards_page_ui.cc:814]: Great catch - thanks!

File: components/brave_rewards/resources/page/components/pageWallet.tsx (hunk 3 above)
Comment (zenparsing) [reply components/brave_rewards/resources/page/components/pageWallet.tsx:127]: Removed.

File: components/brave_rewards/resources/page/components/pageWallet.tsx (hunk 2 above)
Comment (zenparsing) [reply components/brave_rewards/resources/page/components/pageWallet.tsx:52]: Removed.

File: components/brave_rewards/resources/page/brave_rewards_page.tsx (hunk 1 above)
Comment (zenparsing) [reply components/brave_rewards/resources/page/brave_rewards_page.tsx:212]: Reverted this change.

**Review by emerick — APPROVED** [review]
LGTM

File: components/brave_rewards/resources/page/actions/rewards_actions.ts [hunk 5]
```diff
@@ -72,6 +56,8 @@ export const onModalBackupClose = () => action(types.ON_MODAL_BACKUP_CLOSE)
 
 export const onModalBackupOpen = () => action(types.ON_MODAL_BACKUP_OPEN)
 
+export const setBackupCompleted = () => action(types.SET_BACKUP_COMPLETED)
```
Comment (szilardszaloki) [inline components/brave_rewards/resources/page/actions/rewards_actions.ts:59]: Hmm, this should be removed (along with `types.SET_BACKUP_COMPLETED`, and any related code).

File: components/brave_rewards/resources/page/actions/rewards_types.ts [hunk 6]
```diff
@@ -12,15 +12,13 @@ export const enum types {
...
-  ON_CLAIM_PROMOTION = '@@rewards/ON_CLAIM_PROMOTION',
-  ATTEST_PROMOTION = '@@rewards/ATTEST_PROMOTION',
-  RESET_PROMOTION = '@@rewards/RESET_PROMOTION',
   DELETE_PROMOTION = '@@rewards/DELETE_PROMOTION',
   ON_PROMOTION_FINISH = '@@rewards/ON_PROMOTION_FINISH',
   RECOVER_WALLET = '@@rewards/RECOVER_WALLET',
   ON_RECOVER_WALLET_DATA = '@@rewards/ON_RECOVER_WALLET_DATA',
   ON_MODAL_BACKUP_CLOSE = '@@rewards/ON_MODAL_BACKUP_CLOSE',
   ON_MODAL_BACKUP_OPEN = '@@rewards/ON_MODAL_BACKUP_OPEN',
+  SET_BACKUP_COMPLETED = '@@rewards/SET_BACKUP_COMPLETED',
```
Comment (szilardszaloki) [inline components/brave_rewards/resources/page/actions/rewards_types.ts:21]: This also.

File: components/brave_rewards/resources/page/components/pending_contributions_modal.style.ts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+/* Each row is a grid
+    ---------------------------------------------
+    |   | 1(29px) | 2(1fr) |  3(1fr)  | 4(14px) |
+    ---------------------------------------------
+    | 1 |  Icon     Site              |     |_| |
+    | 2 |         | Pending Amount              |
+    | 3 |         | Type  |                Date |
+    ---------------------------------------------
+*/
```
Comment (szilardszaloki) [inline components/brave_rewards/resources/page/components/pending_contributions_modal.style.ts:61]: 👍🏻

**Review by szilardszaloki — APPROVED** [review]
LGTM++, very impressive.

File: components/brave_rewards/resources/page/actions/rewards_actions.ts (hunk 5 above)
Comment (zenparsing) [reply components/brave_rewards/resources/page/actions/rewards_actions.ts]: Fixed.

File: components/brave_rewards/resources/page/actions/rewards_types.ts (hunk 6 above)
Comment (zenparsing) [reply components/brave_rewards/resources/page/actions/rewards_types.ts]: Fixed.

File: components/brave_rewards/resources/page/components/pending_contributions_modal.style.ts (hunk 7 above)
Comment (zenparsing) [reply components/brave_rewards/resources/page/components/pending_contributions_modal.style.ts:61]: Copied from @mkarolin 's excellent work on the Android page!

**Pull request merged**